### PR TITLE
Issue 39640: Error rendering

### PR DESCRIPTION
**Pull request by labkey-adam**

### CODE REVIEW

File: api/src/org/labkey/api/data/DataColumn.java [hunk 1]
```diff
@@ -651,22 +652,22 @@ else if (_inputType.equalsIgnoreCase("checkbox"))
             }
         }
 
-        String errors = getErrors(ctx);
-        if (!StringUtils.isEmpty(errors))
+        HtmlString errors = getErrors(ctx);
```
Comment (labkey-matthewb) [inline api/src/org/labkey/api/data/DataColumn.java:655]: mark getErrors() with @NotNull (or check null != errors)

File: api/src/org/labkey/api/data/RenderContext.java [hunk 2]
```diff
@@ -808,54 +813,54 @@ public void setUseContainerFilter(boolean useContainerFilter)
...
-            String errStr = null;
+            HtmlString errStr;
             if (m instanceof LabKeyError)
             {
                 errStr = ((LabKeyError) m).renderToHTML(getViewContext());
             }
             else
             {
-                errStr = PageFlowUtil.filter(getViewContext().getMessage((MessageSourceResolvable) m), true);
+                errStr = HtmlString.unsafe(PageFlowUtil.filter(getViewContext().getMessage((MessageSourceResolvable) m), true));
```
Comment (labkey-matthewb) [inline api/src/org/labkey/api/data/RenderContext.java]: This is fine.  In the interest of self-documenting code, might be nice to have something like HtmlString.preserveLineBreaks(message)

**Review by labkey-matthewb — CHANGES_REQUESTED** [review]

**Review by labkey-matthewb — APPROVED** [review]

File: api/src/org/labkey/api/data/DataColumn.java (hunk 1 above)
Comment (labkey-adam) [reply api/src/org/labkey/api/data/DataColumn.java:655]: Done

File: api/src/org/labkey/api/data/RenderContext.java (hunk 2 above)
Comment (labkey-adam) [reply api/src/org/labkey/api/data/RenderContext.java]: Added an HtmlString.of() variant that takes a translateWhiteSpaceflag, and migrated a few usages like this. Using a flag matches our approach in PageFlowUtil.filter() and JspBase.h().

**Pull request merged**